### PR TITLE
Add libavif support

### LINKED ISSUE #87: AVIF support using libavif?
This plugin currently depends on libheif, which we cannot currently bring into Fedora. Would it be possible to change this to use libavif instead (which we do have in Fedora)?

This would allow us to bring this plugin into Fedora itself and use it in the KDE spin for our default wallpaper, which has come with animated flavors for a long time on GNOME, but hasn't been available on Plasma in that form.

### PR DESCRIPTION
**Pull request by zzag**
Due to x265 dependency in libheif, some distros can't ship this wallpaper plugin, which is a bummer. On the other hand, there's the AVIF file format, which is equivalent to the HEIF file format but without all the pesky patents.

Draft not tested if it actually works. Also, the wallpaper plugin will need to handle color space information in wallpapers.

closes #87

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +38,7 @@ ecm_find_qmlmodule(QtPositioning 5.12)
 ecm_find_qmlmodule(org.kde.kcm 1.1)
 ecm_find_qmlmodule(org.kde.kirigami 2.10)
 
+find_package(libavif REQUIRED)
 find_package(libheif ${LIBHEIF_MIN_VERSION} REQUIRED)
```
Comment (Conan-Kudo) [inline CMakeLists.txt:42]: You'll need to either drop `libheif` as a dependency entirely or make it optional. If it remains mandatory, we can't ship it.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (zzag) [reply CMakeLists.txt:42]: Yeah, I will most likely drop the libheif code path.

Comment (Conan-Kudo) [reply CMakeLists.txt:42]: Sounds good to me. 👍🏾

**Comment by zzag** [comment]
@karmanyaahm for the record, in version 4, it won't be possible to load dynamic wallpapers created for version 3. One would need to run kdynamicwallpaperbuilder again to get avif wallpapers. kdynamicwallpaperbuilder's metadata format has not been changed.

**Pull request merged**